src/libcore/iter/iterator.rs

```diff
@@ -1696,6 +1736,41 @@ pub trait Iterator {
         self.try_fold(init, move |acc, x| Ok::<B, !>(f(acc, x))).unwrap()
     }
 
+    /// The same as [`fold()`](#method.fold), but uses the first element in the
+    /// iterator as the initial value, folding every subsequent element into it.
+    /// If the iterator is empty, return `None`; otherwise, return the result
+    /// of the fold.
+    ///
+    /// # Example
+    ///
+    /// Find the maximum value:
+    ///
+    /// ```
+    /// fn find_max<I>(iter: I) -> Option<I::Item>
```

Comment:
Aside from the behavior for `Equal`, this is just `Iterator::max` -- maybe a different example can be provided?